assets/src/adaptivity/rules-engine.ts

```diff
@@ -328,6 +328,68 @@ export const getReferencedKeysInConditions = (conditions: any) => {
   return Array.from(references);
 };
 
+export const findReferencedActivitiesInMutateConditions = (conditions: any) => {
```

Comment:
We need to look for the conditions in mutate state as well for variables Required For Evaluation. Just copied the conditions code and refactored it to use for Mutate State. The reason we need is, if there are mutation that changes the question  score, they get evaluated in the rules engine itself hence it needs all the variable to perform the calculation.

Reply:
do we need to have a separate method? or can we not just look at the actions while we're going through the original loop? also if we *do* need to have a separate method, it would be good to change the variable names to reflect what we are working with (actions) instead of conditions so that it is less confusing

Reply:
PR Updated